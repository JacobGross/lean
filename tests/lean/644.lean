import data.list

constant A : Type₁
constant B : Type₁
constant coe : A → B
attribute coe [coercion]
variable f : B → B
variable a : A
set_option pp.coercions true
check f a

constant C : Type₁
variable g : (C → B) → B
variable h : C → A
check g h

definition tst (g : C → B) : Prop := g = h

open bool list

definition bool_to_Prop [coercion] [reducible] (b : bool) : Prop := b = tt

definition bpred_dec [instance] {A : Type} (p : A → bool) : ∀ a, decidable (p a = tt) :=
begin
  intro a,
  eapply (bool.rec_on (p a)),
    right, contradiction,
    left, reflexivity
end

definition negb : bool → bool := bool.rec tt ff

check filter negb [tt, ff, tt, ff]

eval filter negb [tt, ff, tt, ff]

example : filter negb [tt, ff, tt, ff] = [ff, ff] :=
rfl
