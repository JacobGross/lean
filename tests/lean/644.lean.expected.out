f (coe a) : B
g (λ (x : C), coe (h x)) : B
filter (λ (x : bool), bool_to_Prop (negb x)) [tt, ff, tt, ff] : list bool
[ff, ff]
